gazebo/physics/Model.cc

```diff
@@ -77,26 +77,51 @@ void Model::Load(sdf::ElementPtr _sdf)
   // Only <model> is supported right now, <actor> is not supported.
   if (_sdf->GetName() == "model")
   {
-    // If the model is initialized without a containing world, we create an
-    // isolated/standalone DOM object for the world. Otherwise, we'd obtain
-    // the model DOM from the world DOM.
-    const auto *worldDom = this->GetWorld()->GetSDFDom();
-    const std::string modelName = _sdf->Get<std::string>("name");
-    if (nullptr != worldDom && !modelName.empty())
+    const auto modelName = _sdf->Get<std::string>("name");
+
+    // If the model is initialized without a containing parent model or world,
+    // we create an isolated/standalone DOM object for the model. Otherwise,
+    // we'd obtain the model DOM from the parent DOM.
+    BasePtr parentEntity = this->GetParent();
+    if (nullptr != parentEntity && !modelName.empty())
     {
-      this->modelSDFDom = worldDom->ModelByName(modelName);
+      if (boost::dynamic_pointer_cast<Model>(parentEntity))
+      {
+        auto parentDom =
+            boost::dynamic_pointer_cast<Model>(parentEntity)->GetSDFDom();
```

Comment:
nit: if you store the result of `boost::dynamic_pointer_cast<Model>` before the `if` statement, then you can avoid calling it a second time here

Reply:
https://github.com/osrf/gazebo/pull/2824/commits/cbd9853007de3c100ab1da52c99c3af2f45c99d8